### ISSUE TITLE
Regex detecting camelCase functions which are not react components in require-usememo

### ISSUE THREAD

**Issue by michael1997** [arthurgeron/eslint-plugin-react-usememo#9]
The regex to detect component names in require-usememo currently detect function names which are not react components in `require-memo.ts`:

`const componentNameRegex = /^[^a-z]/;`

I propose something like `/^[A-Z][a-zA-Z]*([A-Z][a-zA-Z]*)*$/` to only capture PascalCase component names.

**Comment by arthurgeron-work** [comment]
@michael1997 
Good suggestion, do you mind attaching a screenshot so it can remain documented here?

I've done a shrinked down version of your regexp:  /^([A-Z][a-zA-Z]*)+$/, please provide me some code that replicates the issue so I can test your solution against it.

**Comment by arthurgeron** [comment]
fixed in [v0.1.2](https://github.com/arthurgeron/eslint-plugin-react-usememo/releases/tag/0.1.2)

**Comment by paztis** [comment]
For now any function that start with a capital letter is considered as a React FunctionComponent.
We have too much false positives on our side.

Don't you have a way to verify the arguments of the function for example ? only 1 argument ? This will avoid to detect this for example
```
export const Test = (arg1: string, arg2: string, arg3: string): string => {
    return `${arg1} ${arg2} ${arg3}`;
}
```

In case you have typescript signature you can verify the return type also (JSX.Element)
